### ISSUE TITLE
[HOLD for payment 2023-06-23] [$1000] Allow adding newlines in description for tasks

### ISSUE THREAD

**Issue by thienlnam** [Expensify/App#19135]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Create a new task
2. Add a description
3. Try to add multiple lines 

## Expected Result:
Text input should expand to allow for multiple lines

## Actual Result:
Does not allow for multiple lines

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:** Internal
**Slack conversation:** https://expensify.slack.com/archives/C04QEB4MJEQ/p1684309151082709

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01814d31ef4bb555e0</li>
        <li>Upwork Job ID: 1658903725708025856</li>
        <li>Last Price Increase: 2023-05-17</li>
    </ul>
</details>

**Comment by akinwale** [comment]
> @akinwale don't edit the proposal after some other proposal you clearly edit did after my prospal which is not good practice
> 
> cc @sobitneupane please take this occurrence as well since this is a small issue they should have mentioned completely in first place
> 
> <img alt="Screenshot 2023-05-18 at 12 12 04 AM" width="959" src="https://user-images.githubusercontent.com/47522946/239048413-5ee6f748-1507-4c1e-8939-8902708fa5ac.png">

I posted my proposal and then ran tests before I made my edit.

**Comment by akinwale** [comment]
> @akinwale is it fair to other contributor?

It is ultimately up to C+ / internal engineer to decide. I discovered that I missed a page from my first post when I was testing, and then I updated my proposal. I was rushing to post my initial proposal as well, which is why I missed it in the first place.

**Comment by dukenv0307** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Allow adding newlines in description for tasks
### What is the root cause of that problem?
 New feature
### What changes do you think we should make in order to solve the problem?
In description input in both NewTaskDetailsPage and NewTaskDescriptionPage Component 
https://github.com/Expensify/App/blob/19e683722255d16b38db8ec80dffe3262df78a19/src/pages/tasks/NewTaskDescriptionPage.js#L86-L91
AND
https://github.com/Expensify/App/blob/19e683722255d16b38db8ec80dffe3262df78a19/src/pages/tasks/NewTaskDetailsPage.js#L85-L88

We should add 
```
 autoGrowHeight
 textAlignVertical="top"
```
Like in WorkspaceInviteMessagePage
https://github.com/Expensify/App/blob/19e683722255d16b38db8ec80dffe3262df78a19/src/pages/workspace/WorkspaceInviteMessagePage.js#L178-L189
### What alternative solutions did you explore? (Optional)

**Comment by sobitneupane** [comment]
@youssef-lr @flaviadefaria @thienlnam We have two proposals one to use autogrow as in `WorkspaceInviteMessagePage` and another to use Text Input with fixed number of lines as in "Close Account Page". I don't think `autogrow` carries much value when we have only one or two inputs with enough space.

Do we want the `textinput` to autogrow or have fixed height?

**Comment by thienlnam** [comment]
Good question, we should probably consolidate how we handle these - could you please bring it up in #expensify-open-source? I don't have a strong opinion but maybe someone else does

**Comment by sobitneupane** [comment]
Started a discussion: https://expensify.slack.com/archives/C01GTK53T8Q/p1684489769627739

**Comment by flaviadefaria** [comment]
I'm OoO for 8 days so re-adding the BUG label, but keeping myself assigned. I'll be back at work on May 30th.

**Comment by puneetlath** [comment]
It sounds like we've decided on autogrow @sobitneupane

**Comment by sobitneupane** [comment]
Based on the discussion [here](https://expensify.slack.com/archives/C049HHMV9SM/p1684512067838109?thread_ts=1684480867.153739&cid=C049HHMV9SM), it was decided to use auto-grow.
 
[Proposal](https://github.com/Expensify/App/issues/19135#issuecomment-1551880839) from @akinwale looks good to me.

@akinwale We must also set max-Height along with `autoGrowHeight` similar to `WorkspaceInviteMessagePage`

```
    containerStyles={[styles.workspaceInviteWelcome]} 
```
Please follow `WorkspaceInviteMessagePage` for other required changes.

Changes should be made on `NewTaskDetailsPage `, `NewTaskDescriptionPage` and `TaskDescriptionPage` as well.

🎀👀🎀 C+ reviewed

cc: @youssef-lr

**Comment by sobitneupane** [comment]
I would like to suggest everyone to add a new comment as suggested in guidelines if any update is made in the proposal especially if another proposal is already added or few hours has passed since the proposal has added.

**Comment by dukenv0307** [comment]
@sobitneupane [my proposal](https://github.com/Expensify/App/issues/19135#issuecomment-1551910694) is the first one that covers all cases and suggests updating **like WorkspaceInviteMessagePage**

**Comment by akinwale** [comment]
> @sobitneupane Could you evaluate again? [My proposal](https://github.com/Expensify/App/issues/19135#issuecomment-1551910694) is the first one that covers all cases and suggests updating **like WorkspaceInviteMessagePage**

My proposal was the first to suggest using autogrow which is a solution to the issue, and is something that can be determined by looking at the implementation of the base TextInput component.

Considering newer contributors have less knowledge of the components in the app than more experienced contributors, we don't know about all the intricate details of every component implemented in the system,  so it wouldn't be fair to require having to know exactly how autogrow was handled elsewhere to get an accepted proposal.

**Comment by youssef-lr** [comment]
Not overdue.

**Comment by akinwale** [comment]
@sobitneupane @youssef-lr I have applied to the Upwork job and the PR is ready for review. Thanks.

**Comment by akinwale** [comment]
Still waiting for a response on https://github.com/Expensify/App/pull/19377.

**Comment by puneetlath** [comment]
@akinwale who are you waiting for a response from? What can I do to help you get unblocked?

**Comment by akinwale** [comment]
> @akinwale who are you waiting for a response from? What can I do to help you get unblocked?

Waiting for a response from @youssef-lr who has posted in the [related Slack discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1685087242060519) that he will respond later today.

**Comment by puneetlath** [comment]
@youssef-lr were you able to get what you needed from that discussion?

**Comment by youssef-lr** [comment]
It seems like we decided to keep things as is until we implement a redesign @puneetlath @akinwale

**Comment by puneetlath** [comment]
Got it! In that case, should we close this issue?

**Comment by youssef-lr** [comment]
I think we still want to go with this as it allows us to make task descriptions accept multiple lines. When it comes to the scroll behavior we'll let the task description expand until we redesign it. As demonstrated here: There's a video here which shows what it looks like currently: https://github.com/Expensify/App/pull/19377#issuecomment-1560518276.

**Comment by puneetlath** [comment]
@flaviadefaria assigning back to you to manage.

**Comment by akinwale** [comment]
@youssef-lr Bumping for final review.

Also, I am hoping that the speed bonus can be applied here, considering that the PR in its current form is what I had ready for review a day after I was assigned to the issue, and the delays were due to waiting for responses to clarifications regarding design decisions, and review feedback.

Thanks.

cc: @flaviadefaria

**Comment by thienlnam** [comment]
cc @akinwale @sobitneupane 
Can you both take a look at this issue? https://github.com/Expensify/App/issues/20836, I believe it's from the PR that was merged recently

**Comment by akinwale** [comment]
@flaviadefaria Just checking in here. Looks like the Upwork job set up by automation expired. Thanks.

**Comment by Pujan92** [comment]
Is https://github.com/Expensify/App/issues/21433 issue raised due to this issue PR?

**Comment by flaviadefaria** [comment]
The two comments below need to be addressed:

>cc @akinwale @sobitneupane
Can you both take a look at this issue? https://github.com/Expensify/App/issues/20836, I believe it's from the PR that was merged recently

>Is https://github.com/Expensify/App/issues/21433 issue raised due to this issue PR?


@sobitneupane @akinwale @youssef-lr  can you please check the above? @sobitneupane can you please also fill out the check list? 

@akinwale no need to worry about the job in Upwork once we have everything clarified and ready I'll send you a contract there and link it to this issue.

**Comment by greg-schroeder** [comment]
Yeah, I'm managing https://github.com/Expensify/App/issues/21433 and it seems like this could be directly related? Should that be fixed here or there?

**Comment by sobitneupane** [comment]
> Is https://github.com/Expensify/App/issues/21433 issue raised due to this issue PR?

@greg-schroeder It was a known and pre-existing issue. So, we decided it to be solved separately in another issue.

> Can you both take a look at this issue? https://github.com/Expensify/App/issues/20836, I believe it's from the PR that was merged recently

We are working on the fix for the issue. @akinwale has created [PR](https://github.com/Expensify/App/pull/21371) for the fix.

**Comment by flaviadefaria** [comment]
Thanks @sobitneupane! So I believe you need to fill [this](https://github.com/Expensify/App/issues/19135#issuecomment-1595099659) out and mention that it has introduced a bug right?

**Comment by sakluger** [comment]
Sorry folks, one other new issue, https://github.com/Expensify/App/issues/21449, was reported as stemming from this issue/PR. When you click Enter to add a newline in the description, it doesn't automatically expand the description field height until you start typing. Should this have been addressed in the PR here?

**Comment by akinwale** [comment]
> Sorry folks, one other new issue, #21449, was reported as stemming from this issue/PR. When you click Enter to add a newline in the description, it doesn't automatically expand the description field height until you start typing. Should this have been addressed in the PR here?

No, this is a pre-existing issue with the multiline TextInput as outlined in [this comment](https://github.com/Expensify/App/pull/19377#issuecomment-1557290282). It was ultimately agreed that it would be treated (along with the other behaviour) in a separate issue as may be required.

**Comment by flaviadefaria** [comment]
Ok as soon as @sobitneupane fills [this](https://github.com/Expensify/App/issues/19135#issuecomment-1595099659) out I can proceed with the payment.

**Comment by sakluger** [comment]
@akinwale thanks for clarifying. Did you report those bugs in Slack? I just want to make sure there's no duplicate issue already created.

**Comment by akinwale** [comment]
> @akinwale thanks for clarifying. Did you report those bugs in Slack? I just want to make sure there's no duplicate issue already created.

No, I did not a chance to do so.

**Comment by flaviadefaria** [comment]
Waiting for the issue with the regression to close.

**Comment by flaviadefaria** [comment]
>@flaviadefaria There was https://github.com/Expensify/App/issues/20836. It was handled by https://github.com/Expensify/App/pull/21371. I think we should hold the payment for now.

@sobitneupane it looks like the other GH is live. What should we do here? Are we ready for payment now? If so can oan you fill out the details [here](https://github.com/Expensify/App/issues/19135#issuecomment-1595099659)? Thanks!

**Comment by akinwale** [comment]
> > @flaviadefaria There was #20836. It was handled by #21371. I think we should hold the payment for now.
> 
> @sobitneupane it looks like the other GH is live. What should we do here? Are we ready for payment now? If so can oan you fill out the details [here](https://github.com/Expensify/App/issues/19135#issuecomment-1595099659)? Thanks!

@flaviadefaria @sobitneupane The PR addressing tye regression has been in production for 7 days now, so I believe this is now eligible for payment, based on the guidelines.

Thanks.

**Comment by akinwale** [comment]
@flaviadefaria @sobitneupane Bumping for an update. Would like to close this out. Thanks.

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:


I don't think ButZero Checklist applies here as it is a new feature not a bug.

**Comment by flaviadefaria** [comment]
So just to confirm payments should be:
@sobitneupane $1000 - 50% from regression = $500
@akinwale  $1000 - 50% from regression = $500

Is this correct?

**Comment by akinwale** [comment]
> So just to confirm payments should be: @sobitneupane $1000 - 50% from regression = $500 @akinwale $1000 - 50% from regression = $500
> 
> Is this correct?

@flaviadefaria Payment would be just $1000 for me. No speed bonus of $500 due since there was a regression.

**Comment by sobitneupane** [comment]
> Is this correct?

Yes, I think that's correct. I have requested $500 on Expensify app.

**Comment by akinwale** [comment]
@flaviadefaria Is there anything else I need to do here? I haven't received an Upwork offer yet. Thanks.

**Comment by flaviadefaria** [comment]
@akinwale as you can see [here](https://github.com/Expensify/App/issues/19135#issuecomment-1591050294) this didn't meet the 3 days requirement to qualify for the bonus so the regression is deducted from the $1000, which leave the payment at $500. I'll send you a contract for $500 now.

**Comment by flaviadefaria** [comment]
Offer sent to you @akinwale!

**Comment by akinwale** [comment]
> @akinwale as you can see [here](https://github.com/Expensify/App/issues/19135#issuecomment-1591050294) this didn't meet the 3 days requirement to qualify for the bonus so the regression is deducted from the $1000, which leaves the payment at $500. I'll send you a contract for $500 now.

@flaviadefaria The contributing guidelines do not mention a deduction for regressions for contributors. The only thing mentioned is that a regression makes the issue ineligible for the bonus. The deduction only applies to C+.

Additionally, the reason this issue went beyond the 3-day requirement was due to review delays.

**Comment by akinwale** [comment]
> Offer sent to you @akinwale!

@flaviadefaria Accepted, thanks.

For reference, here's a previous issue I worked on where there was a regression. https://github.com/Expensify/App/issues/18210

The final payments were $1000 / $500 for contributor / C+ respectively.

**Comment by flaviadefaria** [comment]
It doesn't [seem](https://github.com/Expensify/App/issues/18210#issuecomment-1558981857) like the linked GH had a regression. Payment issued so closing this GH!

**Comment by akinwale** [comment]
> It doesn't [seem](https://github.com/Expensify/App/issues/18210#issuecomment-1558981857) like the linked GH had a regression. Payment issued so closing this GH!

I received $500, thanks.

However, as pointed out in [this comment](https://github.com/Expensify/App/issues/18210#issuecomment-1559943824), there was a regression, and I subsequently [refunded $500](https://github.com/Expensify/App/issues/18210#issuecomment-1561968979) from a payment of $1,500, with the C+ [refunding $1,000](https://github.com/Expensify/App/issues/18210#issuecomment-1562046577).

Also, quoting from the [contributing guide](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md):
> If the PR causes a regression at any point within the regression period (starting when the code is merged and ending 7 days after being deployed to production), contributors are not eligible for the 50% bonus.

There's nothing in the document about deducting 50% from the regular payment for a regression, as that only applies to C+. It only states that the contributor would be ineligible for the bonus.

Considering the above, there is still a balance of $500 due. Thanks.

**Comment by flaviadefaria** [comment]
@akinwale you are right, sorry about that! I just sent you another offer for the remaining $500.

**Comment by akinwale** [comment]
> @akinwale you are right, sorry about that! I just sent you another offer for the remaining $500.

I've got it! Thanks.
